--- a/arch/arm64/boot/dts/qcom/sdm845-xiaomi-polaris.dts
+++ b/arch/arm64/boot/dts/qcom/sdm845-xiaomi-polaris.dts
@@ -370,7 +370,7 @@
 };
 
 &cdsp_pas {
-	firmware-name = "qcom/sdm845/Xiaomi/polaris/cdsp.mbn";
+	firmware-name = "qcom/sdm845/polaris/cdsp.mbn";
 
 	status = "okay";
 };
@@ -396,7 +396,7 @@
 };
 
 &gpu_zap_shader {
-	firmware-name = "qcom/sdm845/Xiaomi/polaris/a630_zap.mbn";
+	firmware-name = "qcom/sdm845/polaris/a630_zap.mbn";
 };
 
 &ibb {
@@ -411,7 +411,7 @@
 &ipa {
 	qcom,gsi-loader = "self";
 	memory-region = <&ipa_fw_mem>;
-	firmware-name = "qcom/sdm845/Xiaomi/polaris/ipa_fws.mbn";
+	firmware-name = "qcom/sdm845/polaris/ipa_fws.mbn";
 
 	status = "okay";
 };
@@ -504,8 +504,8 @@
 };
 
 &mss_pil {
-	firmware-name = "qcom/sdm845/Xiaomi/polaris/mba.mbn",
-			"qcom/sdm845/Xiaomi/polaris/modem.mbn";
+	firmware-name = "qcom/sdm845/polaris/mba.mbn",
+			"qcom/sdm845/polaris/modem.mbn";
 
 	status = "okay";
 };
@@ -627,7 +627,7 @@
 		compatible = "qcom,wcn3990-bt";
 
 		/* This path is relative to the qca/ subdir under lib/firmware. */
-		firmware-name = "Xiaomi/polaris/crnv21.bin";
+		firmware-name = "polaris/crnv21.bin";
 
 		vddio-supply = <&vreg_s4a_1p8>;
 		vddxo-supply = <&vreg_l7a_1p8>;
@@ -687,7 +687,7 @@
 };
 
 &venus {
-	firmware-name = "qcom/sdm845/Xiaomi/polaris/venus.mbn";
+	firmware-name = "qcom/sdm845/polaris/venus.mbn";
 
 	status = "okay";
 };
